### PR TITLE
Wait for ClusterInfo to catch up after collection creation

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

Collection creation (with Replication 1) on a Coordinator waited for the ClusterInfo to catch up with a recent enough Plan to see the collection, but did not wait for Current. This can lead to races where accessing a collection right after creation might lead to errors (e.g. getResponsibleServer cannot find servers for a shard of the collection).

- [X] :hankey: Bugfix

### Checklist

- [x] Tests
  - [x] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [x] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.12.0: *(Please link PR)*
  - [ ] Backport for 3.11: *(Please link PR)*
  - [ ] Backport for 3.10: *(Please link PR)*

#### Related Information

Similar fix for Replication 2.0 collection creation: https://github.com/arangodb/arangodb/pull/20929

### CODE REVIEW

File: tests/js/client/shell/shell-cluster-collection.js [hunk 1]
```diff
@@ -989,6 +989,47 @@ function ClusterCollectionSuite () {
       }
     },
 
+    // regression test for https://github.com/arangodb/arangodb/pull/21071
+    testCreateWithSlowCurrentUpdate : function () {
+      let setFailAt;
+      let removeFailAt;
+      if (isServer) {
```
Comment (jsteemann) [inline tests/js/client/shell/shell-cluster-collection.js:996]: `isServer` should always be false here, as the test file is located in `tests/js/client`.

File: tests/js/client/shell/shell-cluster-collection.js [hunk 2]
```diff
@@ -989,6 +989,47 @@ function ClusterCollectionSuite () {
...
+        setFailAt(failurePoint);
+        // create a collection
+        const c = db._create(cn, { numberOfShards: 1, replicationFactor: 1 });
+        const docs = [];
+        for (let i = 0; i < 100; ++i) {
+          docs.push({ _key: "test" + i });
+        }
+        // inserting a document fails if it needs to look up a shard and
+        // ClusterInfo/Current hasn't caught up yet
+        c.insert(docs);
```
Comment (jsteemann) [inline tests/js/client/shell/shell-cluster-collection.js:1027]: wondering what is actually tested here, as this test does not contain any assertions, and will happily silence any exceptions in its `finally` block.
I think the thing tested here is the `insert` after the collection creation, but there should at least be a code comment here to indicate that, if not additional test assertions.

Comment (goedderz) [reply tests/js/client/shell/shell-cluster-collection.js:1027]: The finally block does not silence any exceptions; only a catch would do that. So if the insert throws, the test fails. I've appended the comment to hopefully make it a little clearer.

Comment (jsteemann) [reply tests/js/client/shell/shell-cluster-collection.js:1020]: Thanks, you are obviously right about `finally` vs. `catch`.

**Review by jsteemann — APPROVED** [review]
LGTM

**Pull request merged**